### DISCUSSION TITLE
Is limits ready to work in distributed systems?

### DISCUSSION THREAD

**Discussion by akalex** [alisaifee/limits#187] [Q&A]
I am wondering if it is safe to use `limits` in a distributed system where different processes must operate with shared resources in a mutually exclusive way.

For example, I would like to ensure that multiple processes will not override the counter and thus introduce a chance to go above the limit. Is there a tool that prevents that from happening?

Many thanks for considering my question.

**Comment by alisaifee** [comment]
Yes, as long as you use an external storage (like redis) the rate limits will be applied for a distributed application
